packages/ui/src/views/repo/repo-list/repo-list.tsx

```diff
@@ -41,69 +41,73 @@ export function RepoList({
   query,
   handleResetQuery
 }: PageProps) {
+  const noData = !(repos && repos.length > 0)
+
+  if (noData) {
+    return (
+      <StackedList.Root>
+        <div className="flex min-h-[50vh] items-center justify-center py-20">
+          {hasActiveFilters || query ? (
+            <NoData
+              iconName="no-search-magnifying-glass"
+              title="No search results"
+              description={['Check your spelling and filter options,', 'or search for a different keyword.']}
+              primaryButton={{
+                label: 'Clear search',
+                onClick: handleResetQuery
+              }}
+              secondaryButton={{
+                label: 'Clear filters',
+                onClick: handleResetFilters
+              }}
+            />
+          ) : (
+            <NoData
+              iconName="no-data-folder"
+              title="No repositories yet"
+              description={[
+                'There are no repositories in this project yet.',
+                'Create new or import an existing repository.'
+              ]}
+              primaryButton={{
+                label: 'Create repository',
+                onClick: () => {
+                  /* TODO: add create handler */
+                }
```

Comment:
NIT: lets not pass an empty handler?
```
onClick={noop}
```

Reply:
There are a few scopes of improvement. Just made the changes related to data states for now.